frame/alliance/src/lib.rs

```diff
@@ -0,0 +1,1049 @@
+//! - `vote` - Vote on a motion.
+//! - `close` - Close a motion with enough votes or that has expired.
+//! - `set_rule` - Initialize or update the Alliance's rule by IPFS CID.
+//! - `announce` - Make announcement by IPFS CID.
+//! - `nominate_ally` - Nominate a non-member to become an Ally, without deposit.
+//! - `elevate_ally` - Approve an ally to become a Fellow.
+//! - `kick_member` - Kick a member and slash its deposit.
+//! - `add_blacklist_items` - Add some items, either accounts or websites, to the blacklist.
+//! - `remove_blacklist_items` - Remove some items from the blacklist.
+//!
+//! #### For Members (Only Founders)
+//!
+//! - `veto` - Veto on a motion about `set_rule` and `elevate_ally`.
+//!
+//! #### Root Calls
+//!
+//! - `init_founders` - Initialize the founding members.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+mod types;
+pub mod weights;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_runtime::{
+	traits::{StaticLookup, Zero},
+	RuntimeDebug,
+};
+use sp_std::{convert::TryInto, prelude::*};
+
+use frame_support::{
+	codec::{Decode, Encode, MaxEncodedLen},
+	dispatch::{
+		DispatchError, DispatchResult, DispatchResultWithPostInfo, Dispatchable, GetDispatchInfo,
+		PostDispatchInfo,
+	},
+	ensure,
+	scale_info::TypeInfo,
+	traits::{
+		ChangeMembers, Currency, Get, InitializeMembers, IsSubType, OnUnbalanced,
+		ReservableCurrency,
+	},
+	weights::Weight,
+};
+use pallet_identity::IdentityField;
+
+pub use pallet::*;
+pub use types::*;
+pub use weights::*;
+
+/// Simple index type for proposal counting.
+pub type ProposalIndex = u32;
+
+type UrlOf<T, I> = BoundedVec<u8, <T as pallet::Config<I>>::MaxWebsiteUrlLength>;
+
+type BalanceOf<T, I> =
+	<<T as Config<I>>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+type NegativeImbalanceOf<T, I> = <<T as Config<I>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+/// Interface required for identity verification.
+pub trait IdentityVerifier<AccountId> {
+	/// Function that returns whether an account has an identity registered with the identity
+	/// provider.
+	fn has_identity(who: &AccountId, fields: u64) -> bool;
+
+	/// Whether an account has been deemed "good" by the provider.
+	fn has_good_judgement(who: &AccountId) -> bool;
+
+	/// If the identity provider allows sub-accounts, provide the super of an account. Should
+	/// return `None` if the provider does not allow sub-accounts or if the account is not a sub.
+	fn super_account_id(who: &AccountId) -> Option<AccountId>;
+}
+
+/// The non-provider. Imposes no restrictions on account identity.
+impl<AccountId> IdentityVerifier<AccountId> for () {
+	fn has_identity(_who: &AccountId, _fields: u64) -> bool {
+		true
+	}
+
+	fn has_good_judgement(_who: &AccountId) -> bool {
+		true
```

Comment:
Do we want the default (empty tuple) provider to be permissive?. We could have this be an AllowEverythingVerifier and have the default saying everyone has bad judgement?

Reply:
Then by default nobody could join unless the chain running it has an identity provider. I think in the pattern of `()` essentially "doing nothing" (e.g. when Slash configs are set to `()` they just get burned) then this should also do nothing.

Not a super strong opinion though, I'm also fine with your proposal. Could it be a follow up PR though before this is launched?